test/integration/requests.test.ts

```diff
@@ -256,4 +274,125 @@ describe("/requests", function () {
+        .get("/requests?dev=true")
+        .end(function (err, res) {
+          expect(res).to.have.status(200);
+          expect(res.body.data).to.have.lengthOf(2);
+          expect(res.body.data[0]).to.have.all.keys([
+            "id",
+            "createdAt",
+            "requestedBy",
+            "from",
+            "until",
+            "lastModifiedBy",
+            "message",
+            "updatedAt",
+            "state",
+            "type",
+          ]);
+          done();
+        });
+    });
+
+    it("should return all requests by specific user", function (done) {
+      chai
+        .request(app)
+        .get("/requests?dev=true&requestedBy=testUser")
+        .end(function (err, res) {
+          expect(res).to.have.status(200);
+          expect(res.body.data.every((request: any) => request.requestedBy === "testUser"));
+          done();
+        });
+    });
+
+    it("should return all requests by specific user and state", function (done) {
+      chai
+        .request(app)
+        .get("/requests?dev=true&requestedBy=testUser&state=APPROVED")
+        .end(function (err, res) {
+          expect(res).to.have.status(200);
+          expect(res.body.data.every((e: any) => e.requestedBy === "testUser" && e.state === "APPROVED"));
+          done();
+        });
+    });
+
+    it("should return request of type OOO", function (done) {
+      chai
+        .request(app)
+        .get("/requests?dev=true&type=OOO")
+        .end(function (err, res) {
+          expect(res).to.have.status(200);
+          expect(res.body.data.every((e: any) => e.type === "OOO"));
+          done();
+        });
+    });
+
+    it("should return empty array is no data is found, for specific state and user", function (done) {
+      chai
+        .request(app)
+        .get("/requests?dev=true&requestedBy=testUser2&state=APPROVED")
+        .end(function (err, res) {
+          expect(res).to.have.status(200);
+          expect(res.body).to.have.property("message");
+          expect(res.body.message).to.equal(REQUESTS_NOT_FOUND);
+          expect(res.body.data).to.have.lengthOf(0);
+          done();
+        });
+    });
+
+    // TODO: .skip to be removed
+    it.skip("should return empty array is no data is found", function (done) {
+      chai
+        .request(app)
+        .get("/requests?dev=true&requestedBy=testUserRandom")
+        .end(function (err, res) {
+          expect(res).to.have.status(204);
+          expect(res.body).to.have.property("message");
+          expect(res.body.message).to.equal(REQUESTS_NOT_FOUND);
+          expect(res.body.data).to.have.lengthOf(0);
+          done();
+        });
+    });
+
+    it("should throw error if request id doesn't match", function (done) {
+      chai
+        .request(app)
+        .get("/requests?dev=true&id=ramdonId1")
+        .end(function (err, res) {
+          expect(res).to.have.status(404);
+          expect(res.body.error).to.equal("Not Found");
```

Comment:
Can you please verify that the provided `id` parameter is appropriately handled by the API endpoint

Reply:
it is related to test if yes how we check?